build.gradle

```diff
@@ -1521,6 +1522,91 @@ project(':group-coordinator') {
   srcJar.dependsOn 'processMessages'
 }
 
+project(':test-common') {
+  base {
+    archivesName = "kafka-test-common"
+  }
+
+  dependencies {
+    implementation libs.scalaJava8Compat
+    implementation project(':core')
+    implementation project(':metadata')
+    implementation project(':server')
+    implementation project(':raft')
+    implementation project(':storage')
+    implementation project(':server-common')
+    implementation libs.slf4jApi
+    testImplementation libs.junitJupiter
+    testImplementation libs.mockitoCore
+    testRuntimeOnly libs.slf4jReload4j
+    testRuntimeOnly libs.junitPlatformLanucher
+  }
+
+  sourceSets {
+    main {
+      java {
+        srcDirs = ["src/main/java"]
+      }
+    }
+    test {
+      java {
+        srcDirs = ["src/test/java"]
+      }
+    }
+  }
+
+  checkstyle {
+    sourceSets = []
+  }
+
+  javadoc {
+    enabled = false
+  }
+}
+
+project(':test-common:test-common-api') {
+  base {
+    archivesName = "kafka-test-common-api"
+  }
+
+
+  dependencies {
+    implementation libs.scalaJava8Compat
+    implementation project(':clients')
+    implementation project(':core')
+    implementation project(':group-coordinator')
+    implementation project(':metadata')
+    implementation project(':raft')
+    implementation project(':server')
+    implementation project(':server-common')
+    implementation project(':test-common')
```

Comment:
Seems like we will eventually want to remove these other module dependencies right? Ideally the "test-common-api" is just a minimal set of things like annotations, interfaces, and some configuration classes. Does that match your expectation?

Reply:
> Does that match your expectation?

yes, we can move a part of `RaftClusterInvocationContext` to `test-common` to break the dependencies on other modules.

open https://issues.apache.org/jira/browse/KAFKA-17682